app/modules/tunnel/router_proxy.py

```
@@ -0,0 +1,214 @@
+        out_headers["X-Local-Port"] = str(local_port)
+        return Response(
+            content=response.content,
+            status_code=response.status_code,
+            headers=out_headers,
+        )
+    except httpx.TimeoutException:
+        return JSONResponse(
+            status_code=504,
+            content={"error": "LOCAL_TIMEOUT", "workspace_id": workspace_id, "local_port": local_port},
+        )
+    except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)